httpclient-backend/monix/src/test/scala/sttp/client/httpclient/monix/HttpClientMonixWebSocketTest.scala

```diff
@@ -24,4 +31,48 @@ class HttpClientMonixWebSocketTest extends WebSocketTest[Task] with WebSocketStr
       f: WebSocketFrame.Data[_] => Option[WebSocketFrame]
   ): Observable[WebSocketFrame.Data[_]] => Observable[WebSocketFrame] =
     in => Observable.fromIterable(initial) ++ in.concatMapIterable(m => f(m).toList)
+
+  it should "use pipe to process websocket messages - server-terminated - text accumulator" in {
+    val received = new LinkedBlockingQueue[String]()
+    basicRequest
+      .get(uri"$wsEndpoint/ws/send_and_expect_echo")
+      .response(
+        asWebSocketStreamAlways(streams)(
+          MonixWebSockets.textAccumulator { wholePayload =>
+            received.add(wholePayload)
+            WebSocketFrame.text(wholePayload + "-echo")
+          }
+        )
+      )
+      .send(backend)
+      .map { _ =>
+        received.asScala.toList shouldBe List("test1", "test2", "test3")
+      }
+      .toFuture()
+  }
+
+  it should "use pipe to process websocket messages - client-terminated - text accumulator" in {
```

Comment:
does copying these whole tests add any value? Maybe a single test for the text pipe + some additional unit tests for the conversion itself provide better coverage?

Reply:
hmm I am not sure because to be honest I was totally uncertain whether such a complex combination of observable's operators won't break anything. Also I wanted to see how hard/easily would it be to send additional message in the beginning as in the `client terminated` test.

Reply:
ok, maybe we should then extract the common parts of these tests and just pass in the stream?

Reply:
yes, I think so